todo-mvc/src/main/scala/todomvc/TodoMvc.scala

```diff
@@ -89,6 +89,8 @@ object TodoMvc extends IOWebApp:
   def TodoItem(todo: SignallingRef[IO, Option[Todo]]): Resource[IO, HtmlLiElement[IO]] =
     SignallingRef[IO].of(false).toResource.flatMap { editing =>
       li(
+        // note: the tupling below seems to require explicit type signatures
+        // for `todo` and `editing` so that the scala compiler can help us along with `mapN`
         cls <-- (todo: Signal[IO, Option[Todo]], editing: Signal[IO, Boolean]).mapN { (t, e) =>
```

Comment:
Hmm, maybe we should just rewrite as `map2`.

```suggestion
        cls <-- todo.map2(editing) { (t, e) =>
```

Reply:
Hmm, at least in my code `map2` did not work either. For example, I have something like this:
```
          active.map2(relayUri){
            case (true, theUri) => renderSubscription(theUri,filter)
            case _ => div("")
          }
```
but that gives me:
```
No given instance of type cats.Apply[[A] =>> fs2.concurrent.SignallingRef[cats.effect.IO, A]] was found for parameter F of method map2 in class ApplyOps.
I found:

    cats.Invariant.catsApplicativeForArrow[F, A²]

But method catsApplicativeForArrow in trait InvariantInstances2 does not match type cats.Apply[[A] =>> fs2.concurrent.SignallingRef[cats.effect.IO, A]]

where:    A  is a type variable
          A² is a type variable
```
but this:
```
          (active : Signal[IO,Boolean] ,relayUri : Signal[IO,Uri]).mapN {
            case (true, theUri) => renderSubscription(theUri,filter)
            case _ => div("")
          }
```
compiles just fine.

Reply:
Darn. For the record, here's the relevant issue. I want to take another look at it.
- https://github.com/armanbilge/calico/issues/40